### PR TITLE
chore(docs): add contributor guidelines

### LINKED ISSUE #863: Missing Contributor Guide/Guideline
For new users wishing to contribute to the repository there is no contributor guide or guidelines posted. I am unsure of whether I should just grab an issue and start working, or if there's a procedure to follow if I'd like to contribute to the project. Thanks in advance!

### PR DESCRIPTION
**Pull request by manusa**
## Summary

Add CONTRIBUTING.md with contributor guidelines for the project, aligned
to the containers org conventions and the broader Kubernetes/MCP ecosystem.

Guidelines were drafted after researching CONTRIBUTING.md files from:
- Top containers org projects (podman, buildah, skopeo, crun, toolbox, podlet, ramalama, and others)
- The shared [containers/common CONTRIBUTING.md](https://github.com/containers/common/blob/main/CONTRIBUTING.md)
- Ecosystem projects (mcpchecker, Kuadrant/mcp-gateway, kubernetes-sigs/mcp-lifecycle-operator)
- The [Kubernetes contributor guide](https://www.kubernetes.dev/docs/guide/)

The document covers:
- Code of conduct
- Issue reporting
- Development setup (building, testing)
- Adding new MCP tools
- Pull request workflow
- Conventional commits
- DCO sign-off requirement
- Code review process
- Communication channels

Closes #863

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+
+## Code of Conduct
+
+This project follows the [Containers Community Code of Conduct](https://github.com/containers/common/blob/main/CODE_OF_CONDUCT.md). By participating, you are expected to uphold this code.
+
+## Reporting Issues
+
+Before reporting an issue, check our backlog of [open issues](https://github.com/containers/kubernetes-mcp-server/issues) to see if someone else has already reported it. If so, feel free to add your scenario or additional information to the discussion. Please use a thumbs-up emoji on the original report rather than adding "+1" comments.
+
+If you find a new issue we'd love to hear about it! The most important aspect of a bug report is that it includes enough information for us to reproduce it. Please include as much detail as possible, and try to remove extra details that don't relate to the issue itself. The easier it is for us to reproduce it, the faster it'll be fixed!
```
Comment (Cali0707) [inline CONTRIBUTING.md]: @manusa maybe we can mention something here about feature requests? This seems to read as for bug reports only

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+- Use `testify/suite` for organizing tests into suites.
+- Test the public API only (black-box testing).
+- Use real implementations instead of mocks.
+- Use nested subtests with `s.Run()` and descriptive names.
+- Aim for one assertion per test case.
+- Cover edge cases (nil inputs, empty values, invalid formats).
+
+See [AGENTS.md](AGENTS.md#testing-patterns-and-guidelines) for detailed testing patterns and examples.
+
+### Adding New MCP Tools
```
Comment (Cali0707) [inline CONTRIBUTING.md:71]: Do we want to have any guidance on making _good_ tools? IMO this can be in a separate doc/PR that we reference in here. I'm thinking things like:
1. Don't recreate CRUD operations
2. Have evals showing utility
3. Some tricks we learned from our existing tools (e.g. having actions on a resource be a parameter of the tool)

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+
+## Pull Request Guidelines
+
+No Pull Request (PR) is too small! Typos, additional comments in the code, new test cases, bug fixes, new features, more documentation... it's all welcome!
+
+All PRs should be submitted against the `main` branch. Maintainers will take care of backporting if needed.
+
+While bug fixes can first be identified via an issue, that is not required. It's ok to just open up a PR with the fix, but make sure you include the same information you would have included in an issue, like how to reproduce it.
+
+PRs for new features should include some background on what use cases the new code is trying to address. When possible and when it makes sense, try to break up larger PRs into smaller ones -- it's easier to review smaller code changes. But only if those smaller ones make sense as stand-alone PRs.
```
Comment (Cali0707) [inline CONTRIBUTING.md]: Maybe we should recommend here that for larger new features there is a discussion/issue first so we can agree on the feature before they take time implementing it?

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+
+Keep an eye on the CI results. If something fails, check the logs to see if it's related to your change. If you're unsure, ask in the PR comments.
+
+If changes are requested, amend them into the relevant commit rather than adding extra "fix" commits. Use `git commit --amend` and force-push to your branch. This keeps the git history clean.
+
+## Communication
+
+- [GitHub Issues](https://github.com/containers/kubernetes-mcp-server/issues) -- for bugs and feature requests.
+- [GitHub Discussions](https://github.com/containers/kubernetes-mcp-server/discussions) -- for questions and general discussion.
+- [GitHub Pull Requests](https://github.com/containers/kubernetes-mcp-server/pulls) -- for code contributions.
```
Comment (Cali0707) [inline CONTRIBUTING.md:188]: Maybe add the slack channel here?

File: CONTRIBUTING.md (hunk 2 above)
Comment (manusa) [reply CONTRIBUTING.md:71]: Yes, I was thinking about this too. But I believe this is out of scope for this PR. I think we had several ideas for this that might be scattered around internal discussions or other PR comments.

Shall I remove the section for now or leave a placeholder.
I can create a proper issue for further discussion on how to proceed for adding new/modifying existing tools and toolsets.

Comment (Cali0707) [reply CONTRIBUTING.md:71]: Let's leave the section for now, but maybe capture a discussion or issue to come back to this and make a more complete guide? 

I like what is currently there since it explains how to add a tool/toolset. I think we just need some extra docs to explain how to make good tools/toolsets

Comment (manusa) [reply CONTRIBUTING.md:71]: https://github.com/containers/kubernetes-mcp-server/issues/963
